config/common/cmake/chip_gn_args.cmake

```diff
@@ -0,0 +1,209 @@
+# [Args]:
+#   flags - flags to add
+macro(matter_add_cflags flags)
+    list(APPEND MATTER_CFLAGS_C ${flags})
+endmacro()
+
+# Add items to Matter CXX compiler flags
+# [Args]:
+#   flags - flags to add
+macro(matter_add_cxxflags flags)
+    list(APPEND MATTER_CFLAGS_CC ${flags})
+endmacro()
+
+# Add GNU CPP standard flag to Matter CXX compiler flags
+# [Args]:
+#   std_version - standard version number e.g. 17 for C++17
+macro(matter_add_gnu_cpp_standard std_version)
+    list(APPEND MATTER_CFLAGS_CC -std=gnu++${std_version})
+endmacro()
+
+# Get compiler flags from listed targets.
+# Collect common compile flags and save them in MATTER_CFLAGS
+# Collect C/CXX compile flags and save them in MATTER_CFLAGS_C/MATTER_CFLAGS_CXX
+# [Args]:
+#   targets - list of targets
+macro(matter_get_compiler_flags_from_targets targets)
+    foreach(target ${targets})
+        get_target_common_compile_flags(EXTERNAL_TARGET_CFLAGS ${target})
+        get_lang_compile_flags(EXTERNAL_TARGET_CFLAGS_C ${target} C)
+        get_lang_compile_flags(EXTERNAL_TARGET_CFLAGS_CXX ${target} CXX)
+        list(APPEND MATTER_CFLAGS ${EXTERNAL_TARGET_CFLAGS})
+        list(APPEND MATTER_CFLAGS_C ${EXTERNAL_TARGET_CFLAGS_C})
+        list(APPEND MATTER_CFLAGS_CC ${EXTERNAL_TARGET_CFLAGS_CXX})
+        # Reset between targets
+        set(EXTERNAL_TARGET_CFLAGS "")
+        set(EXTERNAL_TARGET_CFLAGS_C "")
+        set(EXTERNAL_TARGET_CFLAGS_CXX "")
+    endforeach()
+endmacro()
+
+# Generate the common Matter GN arguments.
+# Pass all compilation flags to GN build.
+# Available options are:
+#   DEBUG           Debug build
+#   LIB_SHELL       Add Matter shell library
+#   LIB_PW_RPC      Add Matter PW RPC library
+#   LIB_TESTS       Add Matter unit tests library
+#   DEVICE_INFO_EXAMPLE_PROVIDER Add example device info provider support
+#   PROJECT_CONFIG  Path to the project-specific configuration file
+#
+macro(matter_common_gn_args)
+    set(options)
+    set(oneValueArgs
+        DEBUG 
+        LIB_TESTS 
+        LIB_SHELL 
+        LIB_PW_RPC
+        DEVICE_INFO_EXAMPLE_PROVIDER 
+        PROJECT_CONFIG
+    )
+    set(multiValueArgs)
+
+    cmake_parse_arguments(ARG "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+    convert_list_of_flags_to_string_of_flags(MATTER_CFLAGS MATTER_CFLAGS)
+    convert_list_of_flags_to_string_of_flags(MATTER_CFLAGS_C MATTER_CFLAGS_C)
+    convert_list_of_flags_to_string_of_flags(MATTER_CFLAGS_CC MATTER_CFLAGS_CC)
+
+    if (MATTER_CFLAGS)
+        matter_add_gn_arg_cflags     ("target_cflags"                        ${MATTER_CFLAGS})
+    endif() # MATTER_CFLAGS
+    if (MATTER_CFLAGS_C)
+        matter_add_gn_arg_cflags     ("target_cflags_c"                      ${MATTER_CFLAGS_C})
+    endif() # MATTER_CFLAGS_C
+    if (MATTER_CFLAGS_CC)
+        matter_add_gn_arg_cflags     ("target_cflags_cc"                     ${MATTER_CFLAGS_CC})
+    endif() # MATTER_CFLAGS_CC
+
+    if (ARG_DEBUG)
```

Comment:
Wouldn't it be safer to remove these `if`s and always set the arguments rather than rely that they default to `false`?